### PR TITLE
Add a Composition Function that annotates resources with a quote

### PR DESCRIPTION
**Pull request by negz**
This is mostly useful at the moment to test whether Composition Functions may access the network.

(Also, add a much less interesting function that is slow for adding timeouts.)

### CODE REVIEW

File: xfn-quotable/fnio.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+class DesiredComposite(typing.TypedDict):
+    resource: dict
+    connectionDetails: list[ExplicitConnectionDetail]
+
+
+class DesiredResource(typing.TypedDict):
+    name: str
+    resource: dict
+
+    # TODO(negz): Define these as TypedDicts.
```
Comment (jastang) [inline xfn-quotable/fnio.py:34]: Do TypedDicts use slots? These data containers look like fixed schemas for the most part so I might have used dataclass with slots=True (and I like the other stuff they come with) but it's really just personal preference.

**Review by jastang — COMMENTED** [review]
Saw your request and decided to peek from my mobile but not much else to add otherwise 🙂

File: xfn-quotable/fnio.py (hunk 1 above)
Comment (negz) [reply xfn-quotable/fnio.py:34]: I was unfamiliar with slots! This is actually my first time writing Python 3, haha. I'll look into them, thank you.

File: xfn-quotable/main.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    """Get a quote from quotable.io"""
+    rsp = requests.get("https://api.quotable.io/random")
+    rsp.raise_for_status()
+    j = rsp.json()
+    return (j["author"], j["content"])
+
+
+def read_functionio() -> fnio.FunctionIO:
+    """Read the FunctionIO from stdin."""
+    return yaml.load(sys.stdin.read(), yaml.Loader)
```
Comment (negz) [inline xfn-quotable/main.py:22]: I think I probably want `safe_load` here - sounds like `load` shouldn't be used for untrusted data (which a FnIO may contain).

Comment (negz) [reply xfn-quotable/main.py:22]: Also, seems like https://pypi.org/project/trycast/ could be handy to validate that the input is actually a FunctionIO.

**Comment by negz** [comment]
I'm merging open PRs as-is, since they all target the v1alpha1 Functions implementation. I plan to move them to a defunct directory.

**Pull request merged**